tests/js/server/forgot-tests.js

```diff
@@ -157,6 +180,33 @@ fluid.defaults("fluid.tests.express.user.reset.caseHolder", {
                         }
                     ]
                 },
+                // emailResetForgotRequest
+                {
+                    name: "Testing resetting a user's password using an email address...",
+                    type: "test",
+                    sequence: [
+                        {
+                            func: "{emailResetForgotRequest}.send",
+                            args: [ { email: "{that}.options.testUser.email" } ]
+                        },
+                        // If we catch this event, the timing won't work out to cache the initial response.  We can safely ignore it for now.
+                        //{
+                        //    listener: "fluid.tests.express.user.reset.caseHolder.verifyResponse",
+                        //    event: "{fullResetForgotRequest}.events.onComplete",
+                        //    args: ["{fullResetForgotRequest}", "{fullResetForgotRequest}.nativeResponse", "{arguments}.0", 200]
+                        //},
+                        {
+                            listener: "fluid.tests.express.user.reset.caseHolder.fullResetExtractCodeFromEmailAndReset",
+                            event:    "{testEnvironment}.smtp.mailServer.events.onMessageReceived",
+                            args:     ["{testEnvironment}", "{emailResetResetRequest}"] // testEnvironment, resetRequest
+                        },
+                        {
+                            listener: "fluid.tests.express.user.reset.caseHolder.verifyResponse",
```

Comment:
I assume there is knowledge that the "onComplete" event must definitely follow the "onMessageReceived" event but a comment about the location of this knowledge would be helpful